pkg/kubeadm/master/manifests.go

```diff
@@ -0,0 +1,201 @@
+	CLUSTER_NAME             = "--cluster-name=kubernetes"
+	MASTER                   = "--master=127.0.0.1:8080"
+	HYPERKUBE_IMAGE          = "errordeveloper/hyperquick:master"
+)
+
+// TODO look into what this really means, scheduler prints it for some reason
+//
+//E0817 17:53:22.242658       1 event.go:258] Could not construct reference to: '&api.Endpoints{TypeMeta:unversioned.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:api.ObjectMeta{Name:"kube-scheduler", GenerateName:"", Namespace:"kube-system", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:unversioned.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*unversioned.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]api.OwnerReference(nil), Finalizers:[]string(nil)}, Subsets:[]api.EndpointSubset(nil)}' due to: 'selfLink was empty, can't make reference'. Will not report event: 'Normal' '%v became leader' 'moby'
+
+func WriteStaticPodManifests(params *kubeadmapi.BootstrapParams) error {
+	staticPodSpecs := map[string]api.Pod{
+		// TODO this needs a volume
+		"etcd": componentPod(api.Container{
+			Command: []string{
+				"/bin/sh", // TODO do we really need the parent shell here?
+				"-c",
+				"/usr/local/bin/etcd --listen-peer-urls http://127.0.0.1:2380 --addr 127.0.0.1:2379 --bind-addr 127.0.0.1:2379 --data-dir /var/etcd/data",
+			},
+			Image:         "gcr.io/google_containers/etcd:2.2.1", // TODO parametrise
+			LivenessProbe: componentProbe(2379, "/health"),
+			Name:          "etcd-server",
+			Ports: []api.ContainerPort{
+				{Name: "serverport", ContainerPort: 2380, HostPort: 2380},
+				{Name: "clientport", ContainerPort: 2379, HostPort: 2379},
+			},
+			Resources: componentResources("200m"),
+		}),
+		// TODO bind-mount certs in
+		"kube-apiserver": componentPod(api.Container{
+			Name:  "kube-apiserver",
+			Image: HYPERKUBE_IMAGE,
+			Command: []string{
+				"/hyperkube",
+				"apiserver",
+				"--address=127.0.0.1",
+				"--etcd-servers=http://127.0.0.1:2379",
+				"--cloud-provider=fake", // TODO parametrise
+				"--admission-control=NamespaceLifecycle,LimitRanger,ServiceAccount,PersistentVolumeLabel,ResourceQuota",
+				SERVICE_CLUSTER_IP_RANGE,
+				"--service-account-key-file=/etc/kubernetes/pki/apiserver-key.pem",
+				"--client-ca-file=/etc/kubernetes/pki/ca.pem",
+				"--tls-cert-file=/etc/kubernetes/pki/apiserver.pem",
+				"--tls-private-key-file=/etc/kubernetes/pki/apiserver-key.pem",
+				"--secure-port=443",
+				"--allow-privileged",
+				COMPONENT_LOGLEVEL,
+				"--token-auth-file=/etc/kubernetes/pki/tokens.csv",
+			},
+			VolumeMounts:  []api.VolumeMount{pkiVolumeMount()},
+			LivenessProbe: componentProbe(8080, "/healthz"),
+			Ports: []api.ContainerPort{
+				{Name: "https", ContainerPort: 443, HostPort: 443},
+				{Name: "local", ContainerPort: 8080, HostPort: 8080},
```

Comment:
We are running it in host network namespace already. We can safely get rid
of port declarations, I just copied those from the templates in Kubernetes
Anywhere...

On Tue, 23 Aug 2016, 16:39 Mike Danese, <notifications@github.com> wrote:

> In pkg/kubeadm/master/manifests.go
> <https://github.com/kubernetes/kubernetes/pull/31221#discussion_r75892691>
> :
>
> > +				"--admission-control=NamespaceLifecycle,LimitRanger,ServiceAccount,PersistentVolumeLabel,ResourceQuota",
> > +				SERVICE_CLUSTER_IP_RANGE,
> > +				"--service-account-key-file=/etc/kubernetes/pki/apiserver-key.pem",
> > +				"--client-ca-file=/etc/kubernetes/pki/ca.pem",
> > +				"--tls-cert-file=/etc/kubernetes/pki/apiserver.pem",
> > +				"--tls-private-key-file=/etc/kubernetes/pki/apiserver-key.pem",
> > +				"--secure-port=443",
> > +				"--allow-privileged",
> > +				COMPONENT_LOGLEVEL,
> > +				"--token-auth-file=/etc/kubernetes/pki/tokens.csv",
> > +			},
> > +			VolumeMounts:  []api.VolumeMount{pkiVolumeMount()},
> > +			LivenessProbe: componentProbe(8080, "/healthz"),
> > +			Ports: []api.ContainerPort{
> > +				{Name: "https", ContainerPort: 443, HostPort: 443},
> > +				{Name: "local", ContainerPort: 8080, HostPort: 8080},
>
> Can we just do host network namespace?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/31221/files/e6efd99db925a57763ebb00812715449ca3ea3ef#r75892691>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAPWS61j9bfiFvfCNE1sPDwZyjuUa-Yzks5qixQwgaJpZM4JqsTH>
> .
>